### PR TITLE
RectAreaLight: Updated to new LTC model

### PR DESCRIPTION
**Pull request by WestLangley**
The Linearly Transformed Cosines approach has been updated by the authors, and this PR implements those changes.

1. fresnel is now supported in the illumination model
2. new data sets are provided
3. lookup tables are now parameterized by three.js `roughness` and `sqrt( 1 - cos( theta ) )` rather than `theta` directly

Also, a previous bug has been fixed: specular reflections off of zero-roughness materials are now mirror-perfect.

### CODE REVIEW

**Comment by mrdoob** [comment]
Sweet!

File: src/renderers/shaders/ShaderChunk/bsdfs.glsl [hunk 1]
```diff
@@ -112,64 +112,50 @@ vec3 BRDF_Specular_GGX( const in IncidentLight incidentLight, const in Geometric
...
-	float a = 0.86267 + (0.49788 + 0.01436 * y ) * y;
-	float b = 3.45068 + (4.18814 + y) * y;
+
+	// rational polynomial approximation to theta / sin( theta ) / 2PI
+	float a = 0.8543985 + ( 0.4965155 + 0.0145206 * y ) * y;
+	float b = 3.4175940 + ( 4.1616724 + y ) * y;
 	float v = a / b;
 
-	float theta_sintheta = (x > 0.0) ? v : 0.5 * inversesqrt( 1.0 - x * x ) - v;
+    float theta_sintheta = ( x > 0.0 ) ? v : 0.5 * inversesqrt( max( 1.0 - x * x, 1e-7 ) ) - v;
```
Comment (mrdoob) [inline src/renderers/shaders/ShaderChunk/bsdfs.glsl]: Minor: Indentation change.

Comment (WestLangley) [reply src/renderers/shaders/ShaderChunk/bsdfs.glsl]: Oops... sorry. Fixed.

**Comment by mrdoob** [comment]
Thanks!

**Pull request merged**